src/lib/services/user-service.ts

```diff
@@ -239,50 +240,53 @@ class UserService {
         { username, email, name, password, rootRole }: ICreateUser,
         auditUser: IAuditUser = SYSTEM_USER_AUDIT,
     ): Promise<IUserWithRootRole> {
-        if (!username && !email) {
-            throw new BadDataError('You must specify username or email');
-        }
+        try {
+            if (!username && !email) {
+                throw new BadDataError('You must specify username or email');
+            }
 
-        if (email) {
-            Joi.assert(
+            if (email) {
+                Joi.assert(
+                    email,
+                    Joi.string().email({ ignoreLength: true }),
+                    'Email',
+                );
+            }
+
+            const user = await this.store.insert({
+                username,
                 email,
-                Joi.string().email({ ignoreLength: true }),
-                'Email',
-            );
-        }
+                name,
+            });
 
-        const exists = await this.store.hasUser({ username, email });
```

Comment:
I believe we should still include the `hasUser` check because READ operations are significantly less expensive than WRITE operations. Additionally, in 99.99% of cases, a READ is sufficient, as the concurrency issue described appears to be an extremely rare edge case.

Reply:
I am wondering if you removed it because test failed?

Reply:
Thanks! Yes, let's close it for now. I think we don't need to worry about concurrency issues at the moment.